tests/jballerina-unit-test/src/test/resources/test-src/statements/matchstmt/varbindingpatternmatchpattern/mapping_binding_pattern.bal

```diff
@@ -365,6 +365,43 @@ function testMappingBindingPattern17() {
     assertEquals((), mappingBindingPattern17(1));
 }
 
+type UnaryExpr record {|
+    string operand;
+|};
+
+type TypeCastExpr record {|
+    string oprd;
+|};
+
+type Expr UnaryExpr|TypeCastExpr;
+
+function testTypeNarrowingInMatchStmt1(Expr expr) {
+    match expr {
+        var {oprd: o} => {
```

Comment:
The following examples should also work?
```ballerina
public function main() {
    Expr x = {oprd : "DFD"};
    match x {
        {oprd: var o} => {
            TypeCastExpr c = x; // incompatible types: expected 'TypeCastExpr', found 'Expr'
        }
    }
}
```
```ballerina
public function main() {
    Expr x = {oprd : "DFD"};
    match x {
        {oprd: "DFD"} => {
            TypeCastExpr c = x; //incompatible types: expected 'TypeCastExpr', found 'Expr'
        }
    }
}
```

Reply:
Yes they should. I will have a look. Thanks